source/lac/cuda_vector.cu

```diff
@@ -916,6 +916,14 @@ namespace LinearAlgebra
 
 
     template <typename Number>
+    bool Vector<Number>::all_zero() const
+    {
+      return (linfty_norm() == 0) ? true : false;
```

Comment:
I guess you do this because you don't have element-wise access? Because that requires you to read *all* elements, whereas `all_zero()` can return *as soon as it finds one nonzero element*.